src/optimizer/rules/rewrite_rules.cpp

```diff
@@ -501,10 +501,10 @@ void RewritePullFilterThroughAggregation::Transform(common::ManagedPointer<Abstr
       // (outer_relation.a = (expr))
       correlated_predicates.emplace_back(predicate);
       auto root_expr = predicate.GetExpr();
-      if (root_expr->GetChild(0)->GetDepth() < root_expr->GetDepth()) {
-        new_groupby_cols.emplace_back(root_expr->GetChild(1).Get());
-      } else {
+      if (root_expr->GetChild(0)->GetDepth() > root_expr->GetDepth()) {
```

Comment:
1. This is an awesome PR.
2. Could you please add a comment that links to your issue? Something like "<summary of 1404>, please see #1404 for more details."

Reply:
I borrowed the comment from the `DependentSingleJoinToInnerJoin`. I just need to double check that it's accurate.